lib/index.js

```diff
@@ -108,15 +108,12 @@ api.set = async ({
     // FIXME: may need to allow up to N tokens, one per `clientId` if
     // `clientId` is set
 
-    // if there is an existing, unexpired nonce, reuse it, do not generate
-    // a new one to avoid DoS attack of constantly running bcrypt algorithm
+    // if there is an existing nonce, delete it, and generate a new one
```

Comment:
Is see comments documenting a certain threat have been removed and the opposite of the advice in those comments has now been implemented. Why is this ok? How has this threat been mitigated now or is it no longer considered a threat -- and for what reason?

Reply:
I see a FIXME comment above discussing allowing up to N tokens, one per `clientId` -- was this considered? Is it a viable way to mitigate the threat?

Reply:
@dlongley hmmm..I thought may be the requirements must have changed.. reading the comments on the issue, it says that once the nonce token has already been created we can't resend one. And yes, I didn't consider the comment about `allowing N tokens, one per clientId`. Do you have any suggestions to implement this?

Reply:
Does limiting the number of tokens that a client requests solve the DoS attack?

Reply:
Let's skip the "per clientId" part right now and just implement allowing up to N tokens. I'm not sure what the existing implementation looks like -- please take a look and propose something that you think would work to accomplish it.

> Does limiting the number of tokens that a client requests prevent the DoS attack?

It will cover one of the two attacks we're worried about. We can discuss more over IRC.